### PR TITLE
lightning: add precheck about downstream CDC/PiTR

### LINKED ISSUE #39346: lightning should not use local backend to import to a CDC/PiTR cluster
## Feature Request

**Is your feature request related to a problem? Please describe:**

because it will use ingest to import datam while CDC/PiTR is not aware of ingest

**Describe the feature you'd like:**

Add a precheck that if downstream has running CDC/PiTR task, raise error. But after the lightning is started, CDC/PiTR task can start

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: close #39346

Problem Summary:

### What is changed and how it works?

When found target cluster has running CDC/PiTR tasks, lightning should not start importing.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Add a precheck for lightning that target cluster should not have running CDC/PiTR task
```

### CODE REVIEW

File: br/pkg/lightning/restore/precheck_impl_test.go [hunk 1]
```diff
@@ -581,3 +582,30 @@ func (s *precheckImplSuite) TestTableEmptyCheckBasic() {
...
+	result, err := ci.Check(ctx)
+	s.Require().NoError(err)
+	s.Require().NotNil(result)
+	s.Require().Equal(ci.GetCheckItemID(), result.Item)
+	s.Require().Equal(Critical, result.Severity)
+	s.T().Logf("check result message: %s", result.Message)
+	s.Require().True(result.Passed)
+
+	cli := testEtcdCluster.RandClient()
+	// TODO: what's real data?
```
Comment (lance6716) [inline br/pkg/lightning/restore/precheck_impl_test.go:609]: /cc @joccau @asddongmen 

can you provide some real data for this unit tests?

**Comment by lance6716** [comment]
ptal @joccau @sdojjy

**Review by sdojjy — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]
LGTM

**Review by dsdashun — APPROVED** [review]
LGTM

**Comment by lance6716** [comment]
/hold

forget to close etcd client

**Pull request merged**